### PR TITLE
Add v1.0.0.0-rc.3 release notes

### PR DESCRIPTION
**Pull request by lukekim**

### CODE REVIEW

File: docs/ROADMAP.md [hunk 1]
```diff
@@ -72,8 +63,8 @@ If you have a feature request or suggestion, please [get in touch](https://githu
 | ------------ | ------ |
 | `arrow`      | Stable |
 | `duckdb`     | Stable |
-| `sqlite`     | Stable |
 | `postgresql` | Stable |
+| `sqlite`     | Beta   |
```
Comment (peasee) [inline docs/ROADMAP.md]:
```suggestion
| `sqlite`     | Release Candidate |
```

SQLite is already RC. Table formatting will probably need to get spaced for the other rows.

File: docs/release_notes/v1.0.0-rc.3.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+Spice v1.0.0-rc.3 is the third release candidate for the first major version of Spice.ai OSS. This release continues the focus on production readiness and includes new Iceberg Catalog APIs, DuckDB improvements, and a new Iceberg Data Connector.
+
+## Highlights in v1.0-rc.2
+
+- **Iceberg Catalog APIs:**
+
+- **Iceberg Data Connector:**
+
+- **DuckDB Data Accelerator Improvements:**
+
```
Comment (peasee) [inline docs/release_notes/v1.0.0-rc.3.md:35]: SQLite accelerator has graduated to RC.

File: docs/release_notes/v1.0.0-rc.3.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+## Breaking changes
+
+- API:`v1/datasets/sample` has been removed as it is not particularly useful, can be replicated via SQL, and via the tools endpoint `POST v1/tools/:name`.
+
+## Dependencies
+
+- [OpenTelemetry](https://opentelemetry.io/): Upgraded from 0.26.0 to 0.27.1
+- [Go](https://go.dev/): Upgraded from 1.22 to 1.23 (CLI)
+
```
Comment (peasee) [inline docs/release_notes/v1.0.0-rc.3.md:50]: `datafusion-federation` updated to `ce91d5a6479465ba349f95e3775653e7c27a9185`
`datafusion-table-providers` updated to `19751e31a39c41a454f5b40ef8f1409812769314`
`datafusion` updated to `ffc3ca6bef06a97dbe5c4da1e507ce224511998d`

might need to get updated again if https://github.com/spiceai/spiceai/pull/4064 merges

Comment (phillipleblanc) [reply docs/release_notes/v1.0.0-rc.3.md:50]: I don't think we need to call out those dependency updates, they are effectively our code.

**Review by phillipleblanc — DISMISSED** [review]

**Review by peasee — DISMISSED** [review]

**Review by peasee — DISMISSED** [review]

**Pull request merged**